dev-lang/rust/rust-1.46.0.ebuild

```diff
@@ -198,6 +196,31 @@ src_prepare() {
 
 		"${WORKDIR}/${rust_stage0}"/install.sh --disable-ldconfig \
 			--destdir="${rust_stage0_root}" --prefix=/ || die
+
+		if use prefix; then
+			interpreter=`ldconfig -p | grep ld-linux | awk '{print $NF}' || die "Cannot find your ld.so, why?"`
```

Comment:
we don't use backticks in ebuilds, please use '$(...)' form.
also that pipe does not do what you thing it does, so split it into several steps and don't pipe more than once in single substitution.

also make variables `local`

Reply:
and remove, `why?` part, it's confusing and does not help at all.

Reply:
`ld-linux` is interpreter name only on amd64 and arm64, on other platforms in may be different.

for instance, on ppc64 it's `ld64.so.2`, so that solution is not portable at all.